apps/language_server/lib/language_server/providers/formatting.ex

```diff
@@ -36,13 +36,15 @@ defmodule ElixirLS.LanguageServer.Providers.Formatting do
 
   # If in an umbrella project, the cwd might be set to a sub-app if it's being compiled. This is
   # fine if the file we're trying to format is in that app. Otherwise, we return an error.
-  defp can_format?(file_uri, project_dir) do
+  defp can_format?(file_uri = "file:" <> _, project_dir) do
     file_path = file_uri |> SourceFile.path_from_uri() |> Path.absname()
 
-    not String.starts_with?(file_path, project_dir) or
```

Comment:
The negation that was here seems wrong. @axelson do you have any idea why it was like this?

Reply:
Hmm, it does seem backwards. I'd have to do some investigation but I would guess that it would be related to the fact that previously it wasn't doing a `Path.absname` before. But I'd have to do some more investigation to be more sure. If we do keep a check like it around, then I think we should refactor it to make it a bit more obvious exactly what it is checking (and add some relevant tests).

For reference this is the commit that introduced it: https://github.com/elixir-lsp/elixir-ls/commit/d1437729c0a905cad31f371eac22d4ade1195193

Reply:
Right, this module would use a refactor and better test coverage. Should I create a new issue?